### PR TITLE
Fixed cluster failure when Netwok is not specified.

### PR DESCRIPTION
**Pull request by rohan47**
When Network was not specified in the storagecluster CR, the cluster
creation failed as it was checking for some value that was not defined.

Signed-off-by: rohan47 <rohgupta@redhat.com>

### CODE REVIEW

**Comment by rohan47** [comment]
@leseb @umangachapagain

File: pkg/controller/storagecluster/cephcluster.go [hunk 1]
```diff
@@ -51,10 +51,12 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
...
-	if sc.Spec.Network.IsMultus() {
-		err := validateMultusSelectors(sc.Spec.Network.Selectors)
-		if err != nil {
-			return err
+	if sc.Spec.Network != nil {
+		if sc.Spec.Network.IsMultus() {
+			err := validateMultusSelectors(sc.Spec.Network.Selectors)
+			if err != nil {
+				return err
+			}
```
Comment (umangachapagain) [inline pkg/controller/storagecluster/cephcluster.go:59]: This is same as
https://github.com/openshift/ocs-operator/pull/746/files#diff-d59be10a9ed83fc2cbc7d93e0e430388R245-R248

Is it possible to add a common function? If later we forget that there are 2 places to make changes, it'll cause problems.

Comment (rohan47) [reply pkg/controller/storagecluster/cephcluster.go]: done

File: pkg/controller/storagecluster/cephcluster.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
 		}
 	}
 
-	if sc.Spec.Network.IsMultus() {
+	if isMultus(sc.Spec.Network) {
```
Comment (umangachapagain) [inline pkg/controller/storagecluster/cephcluster.go:54]: Don't we need to add NetworkSpec to CephCluster if it isMultus?

Comment (rohan47) [reply pkg/controller/storagecluster/cephcluster.go:54]: `ensureCephCluster()` returns cephCluster spec only so I didn't put it inside `isMultus()` in order not to change the function definition.

**Review by jarrpa — APPROVED** [review]

**Comment by rohan47** [comment]
> This makes sense, thank you! Please move the unrelated whitespace changes to a different commit.

Moved all the whitespace changes except for this one.
I was not able to revert it.
https://github.com/openshift/ocs-operator/pull/746/files#diff-3ab4add16e230d7d70aa431f83ce9551L616

**Pull request merged**